rfcs/0003-storage.md

```diff
@@ -1,46 +1,108 @@
-2. Any client can use any of these identities to create PoRep proofs. Replicator identities are the CBC encryption keys.
-3. Periodically at a specific PoH count, a replicator that wants to create PoRep proofs signs the PoH hash at that count. That signature is the seed used to pick the block and identity to replicate. A block is 1TB of ledger.
-4. Periodically at a specific PoH count, a replicator submits PoRep proofs for their selected block. A signature of the PoH hash at that count is the seed used to sample the 1TB encrypted block, and hash it. This is done faster than it takes to encrypt the 1TB block with the original identity.
-5. Replicators must submit some number of fake proofs, which they can prove to be fake by providing the seed for the hash result.
-6. Periodically at a specific PoH count, validators sign the hash and use the signature to select the 1TB block that they need to validate. They batch all the identities and proofs and submit approval for all the verified ones.
-7. After #6, replicator client submit the proofs of fake proofs.
+*  Verification requires generating the CBC blocks. That requires space of 2 blocks per identity, and 1 CUDA core per identity for the same dataset. So as many identities at once should be batched with as many proofs for those identities verified concurrently for the same dataset.
+* Validators will randomly sample the set of storage proofs to the set that they can handle, and only the creators of those chosen proofs will be rewarded.
+
+# 6. Validation and Replication Protocol
+
+## 6.1 Definitions
+
+1. Replicator: storage mining client, stores some part of the ledger enumerated in blocks and submits storage proofs to the chain. Not a full-node.
+2. Block: portion of the ledger which is downloaded by the storage client where storage proof data is derived.
+3. Storage proof: a set of sha hash state which is constructed by sampling the encrypted version of the stored ledger block at certain offsets.
+4. Fake storage proof: A proof which has the same format as a storage proof, but the sha state is actually from hashing a known ledger value which the storage
+client can reveal and is also easily verifiable by the network on-chain.
+5. Storage proof confirmation: A transaction by a validator which indicates the set of real and fake proofs submitted by a storage miner.
+6. Storage proof challenge: A transaction from a replicator that verifiably proves that a validator confirmed a fake proof.
+7. Storage proof claim: A transaction from a validator which is after the timeout period given from the storage proof confirmation and which no successful challenges have been observed which rewards the parties of the storage proofs and confirmations.
+8. Storage mining capacity: the number of keys and samples that a storage miner can verify each storage epoch.
```

Comment:
'number of keys and samples that a storage miner can verify each storage epoch' is confusing.  is this referring to the 'number of keys and samples that a replicator can handle during each storage epoch'?

Reply:
Yea, this is the capacity of validators how many proofs they can validate for a given epoch. I changed it to 'storage validation capacity'.